wire-grpc-client/src/main/java/com/squareup/wire/GrpcClient.kt

```diff
@@ -223,16 +223,15 @@ class GrpcClient private constructor(
       responseAdapter: ProtoAdapter<R>
     ) : GrpcMethod<S, R>(path, requestAdapter, responseAdapter) {
       fun invoke(
-        continuation: Continuation<Any>,
         grpcClient: GrpcClient
       ): Any {
-        val (requestChannel, responseChannel) = super.invoke(continuation, grpcClient)
+        val (requestChannel, responseChannel) = super.invoke(grpcClient)
 
-        val coroutineScope = CoroutineScope(continuation.context)
+        val coroutineScope = CoroutineScope(Unconfined)
         return Pair(
             requestChannel,
-            coroutineScope.async {
-              responseChannel.consume { responseChannel.receive() }
+            coroutineScope.async(start = CoroutineStart.ATOMIC) {
```

Comment:
Probably would be a good idea to document why using atomic here.

Reply:
```// We use ATOMIC to guarantee the coroutine started before any cancellation ever happen.```